### ISSUE TITLE
Not handling the action 

### ISSUE THREAD

**Issue by novakw267** [ga-wdi-boston/capstone-project#560]
My buttons all return with a variation of 

```
"Nothing handled the action 'save'. If you did handle the action, this error can be caused by returning true from an action handler in a controller, causing the action to bubble."
```

I am not entirely sure what it means. I would take a guess it has something to do with either my routes or my lack of a controller, but so far everything has been talking to each other without one?? 

router
```
import Ember from 'ember';
import config from './config/environment';

const Router = Ember.Router.extend({
  location: config.locationType,
});

Router.map(function () {
  this.route('sign-up');
  this.route('sign-in');
  this.route('change-password');
  this.route('users');
  this.route('blog', function() {
    this.route('edit');
  });
  this.route('blogs', function() {
    this.route('new');
  });
});

export default Router;
```

blogs/new.hbs which has the create button 
```
<h2>Create blog</h2>

{{blog-form blog='model'
            save='createBlog'
            cancel='cancel'}}

<div class="form-group">
  <label>Title</label>
  {{input type="text"
          class="form-control"
          value="title"
          maxlength="50"
          placeholder="Title"}}
</div>

<div class="form-group">
  <label>Subject</label>
  {{input type="text"
          class="form-control"
          maxlength="75"
          placeholder="Subject"}}
</div>

<div class="form-group">
  <label>Body</label>
  {{input type="text"
          class="form-control"
          value="description"
          maxlength="500"
          placeholder="Body"}}
</div>

<div class="form-group">
  <label>Author</label>
  {{input type="text"
          class="form-control"
          maxlength="50"
          placeholder="Author"}}
</div>

<button type="submit" class="btn btn-primary" {{action 'save'}}>
  Create
</button>

<button class="btn btn-default" {{action "reset"}}>
  Cancel
</button>

{{outlet}}
```

blogs/index.hbs which has my edit and delete buttons in it
```
<div class="info-message">
<h2> Checkout all the sweet blogs! </h2>
</div>
<div class="container">
  {{#each model as |blog|}}
  {{blog-form
     blog=blog
     edit='editBlog'
     delete="deleteBlog"
   }}
    <div class='blog-model'>
      <h2>Title: {{blog.title}}</h2>
      <h3>Subject: {{blog.subject}}</h3>
      <h3>Blog: {{blog.body}}</h3>
      <h3>Author: {{blog.author}}</h3>
      <br>
      <btn class='comment btn btn-primary'>Add Comment</btn>
      <btn class='edit btn btn-primary' {{action 'edit'}}>Edit Blog</btn>
      <btn class='delete btn btn-danger' {{action 'delete'}}>Delete</btn>
    </div>
  {{/each}}
</div>
{{outlet}}
```
blog/edit.hbs
```
{{blog-form
  blog=model
  save='saveBlog'
  cancel='cancel'
}}
{{outlet}}
```

routes/blogs/new.js
```
import Ember from 'ember';

export default Ember.Route.extend({
  model () {
  return this.get('store').createRecord('blog', {});
},
actions: {
  createBlog(blog) {
    blog.save()
    .then(() => this.transitionTo('blogs'));
  },
  cancel () {
    history.back();
  }
}
});
```
routes/blogs/index.js
```
import Ember from 'ember';

export default Ember.Route.extend({
  actions: {
  editBlog (blog) {
    this.transitionTo('blog.edit', blog);
  },
  deleteBlog (blog) {
    blog.destroyRecord();
  },
},
});
```
routes/blog/edit.js
```
import Ember from 'ember';

export default Ember.Route.extend({
  actions: {
  saveBlog(blog) {
    blog.save()
    .then(() => this.transitionTo('blogs'));
  },
  cancel () {
    history.back();
  }
}
});
```
blog-form.js
```
import Ember from 'ember';

export default Ember.Component.extend({
  actions: {
    edit() {
      this.sendAction('edit', this.get('blog'));
    },
    delete() {
      return this.sendAction('delete', this.get('blog'));
    },
    save() {
      this.sendAction('save', this.get('blog'));
    },
    cancel() {
      this.get('blog').rollbackAttributes();
      this.sendAction('cancel');
    }
  }
});
```
Sorry for the wall of text, I just am not sure what part isn't talking to each other. I thought I had covered my bases in that regard.

**Comment by raq929** [comment]
DO NOT APOLOGIZE FOR ADDING CODE. WE ❤️ CODE.

**Comment by jordanallain** [comment]
make sure the names of your actions line up, you have

`save='createBlog'` in `blogs/new.hbs`

and

`save='saveBlog'` in `blog/edit.hbs`

not saying that this is the problem but it can get confusing.

**Comment by novakw267** [comment]
That did not work unfortunately, it also did not change the error either.

This should hopefully be the best interpretation of my flow so far. 

https://www.draw.io/?lightbox=1&highlight=0000ff&edit=_blank&layers=1&nav=1&title=Untitled%20Diagram.xml#Uhttps%3A%2F%2Fraw.githubusercontent.com%2Fnovakw267%2FCapstone-frontend%2Freadme%2FUntitled%2520Diagram.xml

I think I may have nested things incorrectly...

**Comment by novakw267** [comment]
I haven't closed this issue yet, because I am still really unsure of what we actually did because I can not seem to replicate it .

**Comment by novakw267** [comment]
I can't even get the data to display, let alone get anyone of the other buttons to send an action. I am sorta just spinning my wheels here. I also know I have more of these type of features to build, and I unfortunately am not grasping a piece of this.  
I am just trying to get a delete button to work in order to get down the replication of sending the action. 

By my understanding I should be sending the data actions from blog-action.hbs to blog/edit.hbs and then should be calling the deleteBlog from blog/edit.js which should be using the delete from blog-form.js. However I can't even get below the h2 in blog-action to display. 

blog-action.hbs
```
<div class="info-message">
<h2> Checkout all the sweet blogs! </h2>
</div>
<div class="container">
  {{#each model as |blog|}}
    <div class='blog-model'>
      <h2>Title: {{blog.title}}</h2>
      <h3>Subject: {{blog.subject}}</h3>
      <h3>Blog: {{blog.body}}</h3>
      <h3>Author: {{blog.author}}</h3>
      <br>
      <btn class='comment btn btn-primary'>Add Comment</btn>
      <btn class='edit btn btn-primary' {{action 'edit'}}>Edit Blog</btn>
      <btn class='delete btn btn-danger' {{action 'delete'}}>Delete</btn>
    </div>
  {{/each}}
</div>
{{outlet}}
```
blog/edit.hbs
```
{{blog-action blog=model delete="deleteBlog"}}
```
blog/edit.js
```
import Ember from 'ember';

export default Ember.Route.extend({
  model(params) {
    return this.get('store').query('blog', params);
  },
  actions: {
  saveBlog(blog) {
    blog.save()
    .then(() => this.transitionTo('blogs'));
  },
  deleteBlog (blog) {
    console.log("gone!");
    blog.destroyRecord();
  },
  cancel () {
    history.back();
  }
}
});
```
blog-action.js
```
  delete() {
      console.log('almost');
      return this.sendAction('delete', this.get('blog'));
    },
```

**Comment by raq929** [comment]
This was displaying at one point, yes? I thought it was working when I saw it. See what the difference is between this commit and that one?

**Comment by novakw267** [comment]
There isn't anything wrong the code that I am trying to display because it will display when it is in certain files. For example when it is in the template files for blog.hbs or blogs.hbs I can call those in the link-to. But when I put it in a component like it should be, and change to link-to to blog.edit and add the call to the file just like the other file we used, only the h2 text displays. 

for example 

if I put the code in blogs.hbs and call it with 
```
{{#link-to "blogs" class="navbar-brand"}}Blogs{{/link-to}}
```
it will display correctly 

but if I put it in a component such as blog-action and in 
blog/edit.hbs I put
```
{{blog-action }} 
```
and change the link-to 
```
{{#link-to "blog.edit" class="navbar-brand"}}Blogs{{/link-to}}
```

It only displays the 
```
<div class="info-message">
<h2> Checkout all the sweet blogs! </h2>
</div>
``` 
part, so I am unsure if I am just using the wrong file or something?

**Comment by raq929** [comment]
The link-to takes you to the `blogs` route, but `blog/edit` is either a component, or not on the blog route, so it doesn't have the same model hook. What's in the js file for `blog.edit`?

**Comment by payne-chris-r** [comment]
Is this issue still "not handling the action"?

**Comment by novakw267** [comment]
It was, I finally changed the error consistently.

I had to create a new set of more nested components because I didn't have the components I was using in the right place. I had to make them one more file path deeper because it wouldn't see the data I was calling.

Prior to doing this, it was either fully displayed the data by having the html content in a template file, but have a not handling the action error, or I either display almost nothing/ only the template for the main page and get the action to work but not pass any data. 

I am honestly not sure if this really will solve anything but it changed the error I am getting, now I am passing data, along with being on the path getting the action to work. 

I am going to attempt to continue down that path for a bit and then probably open another issue if I can not make any more progress.
